content/courses/native-onchain-development/deserialize-instruction-data.md

```diff
@@ -338,100 +342,97 @@ There's a lot in this example so let's take it one step at a time:
+            msg!("Instruction: Delete Note");
             // Execute program code to delete a note
         }
     }
 }
 ```
 
-For simple programs where there are only one or two instructions to execute, it
-may be fine to write the logic inside the match statement. For programs with
-many different possible instructions to match against, your code will be much
-more readable if the logic for each instruction handler is written in a separate
-function and simply called from inside the `match` statement.
+For simple programs with one or two instructions, placing logic inside the
+`match` statement may suffice. However, for programs with many instructions, it
+is advisable to write the logic for each instruction handler in a separate
+function and call it from within the `match` statement.
 
-### Program file structure
+### Program File Structure
 
-The [Hello World lesson’s](hello-world-program) program was simple enough that
-it could be confined to one file. But as the complexity of a program grows, it's
-important to maintain a project structure that remains readable and extensible.
-This involves encapsulating code into functions and data structures as we've
-done so far. But it also involves grouping related code into separate files.
+The
+[Hello World lesson](/content/courses/native-onchain-development/hello-world-program.md)
+demonstrated a program simple enough to be confined to one file. As program
+complexity grows, maintaining a readable and extensible project structure
+becomes crucial. This involves encapsulating code into functions and data
+structures, and grouping related code into separate files.
 
-For example, a good portion of the code we've worked through so far involves
-defining and deserializing instructions. That code should live in its own file
-rather than be written in the same file as the entry point. By doing so, we
-would then have two files, one with the program entry point and the other with
-the instruction handler:
+For instance, instruction definition and deserialization code should reside in
+its own file, separate from the entry point. This approach might result in two
+files: one for the program entry point and another for the instruction handler.
 
 - **lib.rs**
 - **instruction.rs**
 
-Once you start splitting your program up like this you will need to make sure
-you register all of the files in one central location. We’ll be doing this in
-`lib.rs`. **You must register every file in your program like this.**
+When splitting your program into multiple files, register all files in a central
+location, typically in `lib.rs`. Each file must be registered this way.
 
 ```rust
-// This would be inside lib.rs
+// Inside lib.rs
 pub mod instruction;
 ```
 
-Additionally, any declarations that you would like to be available through `use`
-statements in other files will need to be prefaced with the `pub` keyword:
+Additionally, use the pub keyword to make declarations available for `use`
```

Comment:
```suggestion
Additionally, use the `pub` keyword to make declarations available for `use`
```

You should do this elsewhere too. Eg like "clear everything in the
current `lib.rs`" below.

Reply:
I have highlighted both syntaxes.